File: take_screenshots.py
#!/usr/bin/env python3
"""Take screenshots of all Workflow DREETS pages for docs.php"""
import os
import sys
import time
import subprocess
import signal
from playwright.sync_api import sync_playwright

PHP_BIN = "/tmp/my-project/bin/php/bin/php"
DOCROOT = "/home/z/my-project/formulaire-dematerialise"
PORT = 8765
OUT_DIR = os.path.join(DOCROOT, "docs", "screenshots")
os.makedirs(OUT_DIR, exist_ok=True)

HEADERS = {
    "X-Test-Mode": "1",
    "X-Test-User": "olivier.noblanc@dreets.gouv.fr"
}

PAGES = [
    ("01_index_agent", "/index.php"),
    ("02_index_admin", "/index.php"),
    ("03_form_onboarding", "/form.php?f=onboarding"),
    ("04_form_outboarding", "/form.php?f=outboarding"),
    ("05_my_submissions", "/my_submissions.php"),
    ("06_my_validations", "/my_validations.php"),
    ("07_dashboard", "/dashboard.php"),
    ("08_monitoring", "/monitoring.php"),
    ("09_admin_access", "/admin_access.php"),
    ("10_admin_forms", "/admin_forms.php"),
    ("11_admin_alerts", "/admin_alerts.php"),
    ("12_admin_settings", "/admin_settings.php"),
    ("13_docs", "/docs.php"),
    ("14_changelog", "/changelog.php"),
    ("15_validate", "/my_validations.php"),
    ("16_submission_view", "/my_submissions.php"),
    ("17_form_preview", "/admin_forms.php"),
]

def take_screenshots():
    # Start PHP server
    print(f"Starting PHP server on port {PORT}...")
    php_proc = subprocess.Popen(
        [PHP_BIN, "-S", f"0.0.0.0:{PORT}", "-t", DOCROOT],
        cwd=DOCROOT,
        stdout=subprocess.PIPE,
        stderr=subprocess.PIPE
    )
    time.sleep(3)
    
    # Verify server is running
    import urllib.request
    try:
        req = urllib.request.Request(f"http://127.0.0.1:{PORT}/index.php")
        req.add_header("X-Test-Mode", "1")
        req.add_header("X-Test-User", "olivier.noblanc@dreets.gouv.fr")
        r = urllib.request.urlopen(req, timeout=5)
        print(f"Server is running (HTTP {r.status})")
    except Exception as e:
        print(f"Server failed to start: {e}")
        php_proc.kill()
        sys.exit(1)
    
    try:
        with sync_playwright() as p:
            browser = p.chromium.launch(headless=True, args=["--no-sandbox", "--disable-setuid-sandbox"])
            context = browser.new_context(
                viewport={"width": 1280, "height": 900},
                extra_http_headers=HEADERS
            )
            
            for i, (name, path) in enumerate(PAGES):
                page = context.new_page()
                url = f"http://127.0.0.1:{PORT}{path}"
                print(f"[{i+1}/{len(PAGES)}] Capturing {name} -> {url}")
                
                try:
                    page.goto(url, wait_until="networkidle", timeout=20000)
                    time.sleep(1.5)
                    page.evaluate("window.scrollTo(0, 0)")
                    time.sleep(0.3)
                    
                    outfile = os.path.join(OUT_DIR, f"{name}.png")
                    page.screenshot(path=outfile, full_page=False)
                    print(f"  ✓ Saved {outfile} ({os.path.getsize(outfile)} bytes)")
                    
                except Exception as e:
                    print(f"  ✗ Error: {e}")
                
                page.close()
            
            browser.close()
    finally:
        php_proc.terminate()
        php_proc.wait(timeout=5)
        print("PHP server stopped")
    
    print(f"\nDone! All screenshots saved to {OUT_DIR}")

if __name__ == "__main__":
    take_screenshots()

File: take_screenshots.js
const { chromium } = require('playwright');
const fs = require('fs');
const path = require('path');

const BASE_URL = 'http://localhost:8899';
const SCREENSHOT_DIR = '/home/z/my-project/formulaire-dematerialise/docs/screenshots';

const PAGES = [
    { path: '/index.php', file: '01_index_agent.png', title: 'Accueil agent' },
    { path: '/index.php', file: '02_index_admin.png', title: 'Accueil admin' },
    { path: '/form.php?f=onboarding', file: '03_form_onboarding.png', title: 'Formulaire onboarding' },
    { path: '/form.php?f=acces_si', file: '04_form_acces_si.png', title: 'Formulaire accès SI' },
    { path: '/my_submissions.php', file: '05_my_submissions.png', title: 'Mes demandes' },
    { path: '/my_validations.php', file: '06_my_validations.png', title: 'Mes validations' },
    { path: '/dashboard.php', file: '07_dashboard.png', title: 'Supervision' },
    { path: '/monitoring.php', file: '08_monitoring.png', title: 'Surveillance' },
    { path: '/admin_access.php', file: '09_admin_access.png', title: 'Accès admin' },
    { path: '/admin_forms.php', file: '10_admin_forms.png', title: 'Gestion formulaires' },
    { path: '/admin_alerts.php', file: '11_admin_alerts.png', title: 'Alertes' },
    { path: '/admin_settings.php', file: '12_admin_settings.png', title: 'Paramètres' },
    { path: '/docs.php', file: '13_docs.png', title: 'Documentation' },
    { path: '/changelog.php', file: '14_changelog.png', title: 'Changelog' },
    { path: '/health.php', file: '15_health.png', title: 'Santé système' },
    { path: '/backup.php', file: '16_backup.png', title: 'Sauvegarde' },
    { path: '/rgpd.php', file: '17_rgpd.png', title: 'RGPD' },
    { path: '/stats.php', file: '18_stats.png', title: 'Statistiques' },
];

async function runScreenshots() {
    // Ensure directory exists
    if (!fs.existsSync(SCREENSHOT_DIR)) {
        fs.mkdirSync(SCREENSHOT_DIR, { recursive: true });
    }

    const browser = await chromium.launch({ headless: true });
    const context = await browser.newContext({
        viewport: { width: 1280, height: 900 },
        extraHTTPHeaders: {
            'X-Test-Mode': '1',
            'X-Test-User': 'test.agent@dreets.gouv.fr'
        }
    });

    let ok = 0;
    let fail = 0;
    const errors = [];

    for (const page of PAGES) {
        const p = await context.newPage();
        try {
            const response = await p.goto(`${BASE_URL}${page.path}`, {
                waitUntil: 'networkidle',
                timeout: 15000
            });

            const status = response ? response.status() : 0;

            // Check if response is JSON (test mode for some pages)
            const contentType = response ? response.headers()['content-type'] || '' : '';
            if (contentType.includes('application/json')) {
                console.log(`  ⏭️  ${page.file} — ${page.title} (JSON en mode test, ignoré)`);
                await p.close();
                continue;
            }

            if (status !== 200) {
                throw new Error(`HTTP ${status}`);
            }

            // Wait a bit for rendering
            await p.waitForTimeout(500);

            const screenshotPath = path.join(SCREENSHOT_DIR, page.file);
            await p.screenshot({ path: screenshotPath, fullPage: false });

            const fileSize = fs.statSync(screenshotPath).size;
            console.log(`  ✅ ${page.file} — ${page.title} (${(fileSize / 1024).toFixed(0)} Ko)`);
            ok++;

        } catch (err) {
            console.log(`  ❌ ${page.file} — ${page.title}: ${err.message}`);
            fail++;
            errors.push(`${page.file}: ${err.message}`);
        } finally {
            await p.close();
        }
    }

    await browser.close();

    console.log(`\n═══════════════════════════════════════════════════════════`);
    console.log(`  SCREENSHOTS: ${ok} OK / ${fail} échoué(s) / ${ok + fail} total`);
    console.log(`═══════════════════════════════════════════════════════════`);
    if (errors.length > 0) {
        console.log(`\nÉchecs:`);
        errors.forEach(e => console.log(`  • ${e}`));
    }

    process.exit(fail > 0 ? 1 : 0);
}

runScreenshots().catch(err => {
    console.error('Error:', err);
    process.exit(1);
});
